pkg/river/token/builder/builder.go

```diff
@@ -216,13 +216,6 @@ func (b *Body) encodeField(prefix []string, field rivertags.Field, fieldValue re
 		fullName := mergeStringSlice(prefix, field.Name)
 
 		switch {
-		case fieldValue.IsZero():
-			// It shouldn't be possible to have a required block which is unset, but
-			// we'll encode something anyway.
-			inner := NewBlock(fullName, "")
-			inner.body.SetValueOverrideHook(b.valueOverrideHook)
-			b.AppendBlock(inner)
-
```

Comment:
Here a zero struct is encoded as empty block in River. When it's decoded, the defaults will be applied, so we wouldn't get the same value as we encoded. 

Luckily all we need to do is remove this case and the other cases will handle it correctly, taking defaults into account.